src/cyverse_groups_client/core.clj

```diff
@@ -136,15 +154,19 @@
                      {:query-params {:user user}
                       :as           :json})))
 
-  (revoke-folder-privilege [_ user name subject privilege]
-    (:body (http/delete (build-url base-url "folders" name "privileges" subject privilege)
-                        {:query-params {:user user}
-                         :as           :json})))
-
-  (grant-folder-privilege [_ user name subject privilege]
-    (:body (http/put (build-url base-url "folders" name "privileges" subject privilege)
-                     {:query-params {:user user}
-                      :as           :json})))
+  (revoke-folder-privilege [_ user folder-name subject privilege]
+    (let [privilege (name privilege)]
+      (validate-folder-privilege privilege)
+      (:body (http/delete (build-url base-url "folders" folder-name "privileges" subject privilege)
+                          {:query-params {:user user}
+                           :as           :json}))))
+
+  (grant-folder-privilege [_ user folder-name subject privilege]
+    (let [privilege (name privilege)]
+      (validate-folder-privilege privilege)
```

Comment:
Is `privilege` already a keyword? It looks like it's being converted into a string here, but then back to a keyword in `validate-folder-privilege`, then `name` is called on it again in `invalid-privilege-msg`?